### PR TITLE
feat: optimize shell data fetching with focused queries

### PR DESCRIPTION
**Pull request by sean-brydon**
## What does this PR do?

Optimizes shell component data fetching by splitting the heavy `me.get` query (30+ fields) into focused queries that return only what each component needs, and consolidating team billing queries.

**New focused endpoints:**
- `viewer.me.theme` - returns only brandColor, darkBrandColor, appTheme (3 fields vs 30+)
- `viewer.me.avatar` - returns only avatar, avatarUrl, name, username
- `viewer.me.profiles` - returns only profiles array
- `viewer.me.premium` - returns only isPremium boolean
- `viewer.teams.billingStatus` - consolidates hasTeamPlan + hasActiveTeamPlan into single query
- `viewer.teams.inviteCount` - returns count only instead of full membership objects

**Updated shell components:**
- `useAppTheme` → uses `useMeTheme`
- `UserDropdown` → uses `useMeAvatar`
- `ProfileDropdown` → uses `useMeProfiles`
- `useHasPaidPlan` → uses `billingStatus` + `premium`
- `TeamInviteBadge` → uses `inviteCount`

Existing `me.get` and team queries are preserved for backward compatibility.

## Updates since last revision

- Fixed `UserDropdown.test.tsx` to mock `useMeAvatar` instead of `useMeQuery` (test was failing because the component now uses the new focused hook)
- Added `select: { plan: true }` to `platformBilling.findUnique` query in `billingStatus.handler.ts` to fetch only the required field (addresses Cubic AI feedback)

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). N/A - internal optimization only.
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

1. Navigate to any page that uses the shell (e.g., `/event-types`, `/bookings`)
2. Verify the shell renders correctly with user avatar, theme colors, and profile dropdown
3. Check the team invite badge shows correctly if you have pending invites
4. Verify the "Upgrade" prompts work correctly based on billing status

**Key areas to verify:**
- User dropdown shows correct avatar and name
- Profile switcher works (if enabled)
- Theme/brand colors apply correctly
- Team invite badge shows/hides appropriately

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have checked if my changes generate no new warnings
- [ ] My PR is not too large (>500 lines or >10 files) - 19 files changed but mostly new focused handlers

## Human Review Checklist

1. **`useTeamInvites` return type changed** from `{ listInvites }` to `{ count }` - verify no other consumers exist in the codebase
2. **`billingStatus.handler.ts`** consolidates logic from two handlers - please review the billing status checking logic carefully for edge cases
3. **Backward compatibility** - old endpoints (`me.get`, `hasTeamPlan`, `hasActiveTeamPlan`, `listInvites`) are preserved
4. **Test coverage** - verify if other components using new hooks have adequate test coverage

---

Link to Devin run: https://app.devin.ai/sessions/a743c9483dff4791b3993b65300cbe8a
Requested by: @sean-brydon

### CODE REVIEW

File: packages/trpc/server/routers/viewer/me/theme.handler.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  const { user: sessionUser } = ctx;
+
+  const user = await prisma.user.findUniqueOrThrow({
+    where: { id: sessionUser.id },
+    select: {
+      brandColor: true,
+      darkBrandColor: true,
+      appTheme: true,
+    },
+  });
```
Comment (eunjae-lee) [inline packages/trpc/server/routers/viewer/me/theme.handler.ts]: let's convert this to a repository

File: packages/trpc/server/routers/viewer/teams/billingStatus.handler.ts [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+      isTrial: false,
+    };
+  }
+
+  let isTrial = false;
+
+  // Check if user has at least one membership with an active plan
+  for (const team of teams) {
+    if (team.isPlatform && team.isOrganization) {
+      const platformBilling = await prisma.platformBilling.findUnique({ where: { id: team.id } });
```
Comment (eunjae-lee) [inline packages/trpc/server/routers/viewer/teams/billingStatus.handler.ts]: repository

File: packages/trpc/server/routers/viewer/teams/inviteCount.handler.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+type InviteCountOptions = {
+  ctx: {
+    user: NonNullable<TrpcSessionUser>;
+  };
+};
+
+export const inviteCountHandler = async ({ ctx }: InviteCountOptions) => {
+  const userId = ctx.user.id;
+
+  const count = await prisma.membership.count({
```
Comment (eunjae-lee) [inline packages/trpc/server/routers/viewer/teams/inviteCount.handler.ts]: repo

File: packages/trpc/react/hooks/useMeAvatar.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import { trpc } from "../trpc";
+
+/**
+ * Focused hook for fetching only avatar-related user data.
+ * Returns: avatar, avatarUrl, name, username
+ * Use this instead of useMeQuery when you only need avatar/identity data.
+ */
+export function useMeAvatar() {
+  return trpc.viewer.me.avatar.useQuery(undefined, {
```
Comment (eunjae-lee) [inline packages/trpc/react/hooks/useMeAvatar.ts:9]: what if we give these little hooks a longer cache expiry?

**Review by eunjae-lee — COMMENTED** [review]
looks really good! i haven't tested but will do it later today

File: packages/trpc/server/routers/viewer/teams/billingStatus.handler.ts [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+  const plan = await billingPlanService.getUserPlanByMemberships(memberships);
+
+  // Get active team plan status
+  const whereClause: Prisma.MembershipWhereInput = { userId, accepted: true };
+
+  if (input?.ownerOnly) {
+    whereClause.role = "OWNER";
+  }
+
+  const teams = await MembershipRepository.findAllAcceptedTeamMemberships(userId, whereClause);
```
Comment (eunjae-lee) [inline packages/trpc/server/routers/viewer/teams/billingStatus.handler.ts:49]: can we hide this prisma where input clause by abstracting it into the repository?

File: packages/trpc/server/routers/viewer/teams/billingStatus.handler.ts [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+      const platformBilling = await platformBillingRepository.findPlanByTeamId(team.id);
+      if (platformBilling && platformBilling.plan !== "none" && platformBilling.plan !== "FREE") {
+        return {
+          hasTeamPlan: !!hasTeamPlan,
+          plan,
+          isActive: true,
+          isTrial: false,
+        };
+      }
+    }
```
Comment (eunjae-lee) [inline packages/trpc/server/routers/viewer/teams/billingStatus.handler.ts:76]: I wonder if this can be achieved with a single query

**Review by eunjae-lee — CHANGES_REQUESTED** [review]
Looks good overall. Made a few suggestions.
